src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1169,6 +1169,28 @@ public void updateFeedbackSession(FeedbackSessionAttributes newSession)
 
         fsDb.updateFeedbackSession(newSession);
     }
+
+    public void updateSessionResponseRateForDeletedStudentResponse(FeedbackResponseAttributes deletedResponse)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        String email = deletedResponse.giverEmail;
+        String sessionName = deletedResponse.feedbackSessionName;
+        String courseId = deletedResponse.courseId;
+
+        if (noRemainingResponsesFromStudent(email, sessionName, courseId)) {
+            deleteStudentFromRespondantList(email, sessionName, courseId);
+        }
+    }
+
+    private boolean noRemainingResponsesFromStudent(String studentEmail, String sessionName, String courseId) {
+        List<FeedbackResponseAttributes> responses = frLogic.getFeedbackResponsesForSession(sessionName,
```

Comment:
I think this is fine for now, as we should try to keep the PR small, but let's make an issue when this is about to be merged to improve performance here.